### PR TITLE
Upgrade to Airflow 1.10.14

### PR DESCRIPTION
**Pull request by jdddog**
This pull request upgrades the Observatory to use Airflow 1.10.14, which is the [second step for upgrading to Airflow 2.0](https://airflow.apache.org/docs/apache-airflow/2.0.0/upgrading-to-2.html#step-2-upgrade-to-airflow-1-10-14-a-k-a-our-bridge-release) (since we already use Python 3).

Once this pull request is merged we should make the changes that are required to DAGs so that they will run in Airflow 2.0, since a lot of features from Airflow 2.0 have been backported to this release, i.e. Airflow 2.0 compatible DAGs work in this version. See the linked page for more details.

### CODE REVIEW

**Review by aroelo — DISMISSED** [review]
Hey Jamie,

Looks good! I've added a couple of small comments.
Also, the docs need to be updated to include the 'secret_key', both for the development and terraform environments.

Not directly related to this PR, but for 2.0. When I tried airflow 2.0 with the telescope template, it looks like some additional changes are required as mentioned in the Slack chat.

Are you planning to just run the Upgrade check scripts and then upgrade to 2.0, or do you want to stay on 1.10.14 for a while and test/run the DAGs with this version first?

File: observatory-platform/requirements.txt [hunk 1]
```diff
@@ -1,26 +1,24 @@
...
+tldextract==3.1.*
+docker-compose==1.28.*
 click==7.1.*
 oauth2client==1.5.2
-importlib-metadata==1.7.*
 httpretty==1.0.*
 pbr==5.5.*
 redis==3.5.*
-importlib-metadata==1.7.*
+requests==2.20.0
```
[the remark this answers is not in the document]
Comment (jdddog) [reply observatory-platform/requirements.txt:23]: There are a few errors that occurred when installing packages:
```bash
ERROR: requests 2.24.0 has requirement idna<3,>=2.5, but you'll have idna 3.1 which is incompatible.
ERROR: flask-jwt-extended 3.25.0 has requirement PyJWT<2.0,>=1.6.4, but you'll have pyjwt 2.0.1 which is incompatible.
ERROR: apache-airflow 1.10.14 has requirement requests<2.24.0,>=2.20.0; python_version >= "3.0", but you'll have requests 2.24.0 which is incompatible.
ERROR: flask-appbuilder 2.3.4 has requirement marshmallow<3.0.0,>=2.18.0, but you'll have marshmallow 3.10.0 which is incompatible.
```

So I've pinned the following versions:
```
requests==2.20.0
pyjwt==1.7.1
idna==2.7
marshmallow==2.21.0
```

File: observatory-platform/observatory/platform/terraform/variables.tf [hunk 2]
```diff
@@ -14,6 +14,7 @@ ui_user_email: the email address for the Apache Airflow UI admin user.
 EOF
   type = object({
     fernet_key = string
+    secret_key = string
     ui_user_email = string
     ui_user_password = string
   })
```
[the remark this answers is not in the document]
Comment (jdddog) [reply observatory-platform/observatory/platform/terraform/variables.tf:21]: Thanks, I've added the description.

**Comment by jdddog** [comment]
> Hey Jamie,
> 
> Looks good! I've added a couple of small comments.
> Also, the docs need to be updated to include the 'secret_key', both for the development and terraform environments.
> 
> Not directly related to this PR, but for 2.0. When I tried airflow 2.0 with the telescope template, it looks like some additional changes are required as mentioned in the Slack chat.
> 
> Are you planning to just run the Upgrade check scripts and then upgrade to 2.0, or do you want to stay on 1.10.14 for a while and test/run the DAGs with this version first?

Thanks Aniek, I've made the changes.

I'm thinking that if we want to use any Airflow 2.0 features for the DAGs, to make them to the 1.10.14 version and test them. Then as another pull request we can run the upgrade check scripts and upgrade to 2.0.

**Pull request merged**